src/java.base/share/classes/java/lang/invoke/InnerClassLambdaMetafactory.java

```diff
@@ -204,12 +215,102 @@ public InnerClassLambdaMetafactory(MethodHandles.Lookup caller,
     }
 
     private static String lambdaClassName(Class<?> targetClass) {
+        return createNameFromTargetClass(targetClass) + counter.incrementAndGet();
+    }
+
+    private static String createNameFromTargetClass(Class<?> targetClass) {
         String name = targetClass.getName();
         if (targetClass.isHidden()) {
             // use the original class name
             name = name.replace('/', '_');
         }
-        return name.replace('.', '/') + "$$Lambda$" + counter.incrementAndGet();
+        return name.replace('.', '/') + "$$Lambda$";
+    }
+
+    /**
+     * Calculate hash value of the given String in the same manner as the
+     * java.lang.StringUTF16#hashCode does with difference that hash value
+     * is long instead of int.
+     *
+     * @param name String for which method calculates hash value for
+     *
+     * @return a hash value for the given String
+     * */
+    private String fixedSizeStringHash(String name) {
+        long h = 0;
+        int length = name.length();
+        for (int i = 0; i < length; i++) {
+            h = 31 * h + name.charAt(i);
+        }
+
+        StringBuilder stable = new StringBuilder(Long.toString(Math.abs(h), Character.MAX_RADIX));
+        int stableLength = stable.length();
+
+        // We want all the hashed names to be of the same length. We pad some of them with the character 'a'
+        // to achieve this.
+        while (stableLength != targetLengthForStableName) {
+            stable.append(paddingCharacter);
+            stableLength++;
+        }
+
+        return stable.toString();
+    }
+
+    private String stableLambdaNameHash(String name) {
+        StringBuilder[] hashFragments = new StringBuilder[]{new StringBuilder(), new StringBuilder()};
+        int n = name.length();
+
+        for (int i = 0; i < n; i++) {
+            hashFragments[i % hashFragments.length].append(name.charAt(i));
+        }
+
+        StringBuilder stableNameHash = new StringBuilder();
+        for (StringBuilder sb : hashFragments) {
+            stableNameHash.append(fixedSizeStringHash(sb.toString()));
+        }
+
+        return stableNameHash.toString();
+    }
+
+    /**
+     * Creating stable name for lambda class.
+     * Parameters that are used to create stable name
+     * are a superset of the parameters that are used in
+     * {@link java.lang.invoke.LambdaProxyClassArchive#addToArchive}
+     * to store lambdas.
+     *
+     * @return a stable name for the created lambda class.
+     */
+    private String stableLambdaClassName(Class<?> targetClass) {
+        String name = createNameFromTargetClass(targetClass);
+
+        StringBuilder stableName = new StringBuilder().append(interfaceMethodName);
```

Comment:
`StringBuilder hashData` might be a bit more fitting

Reply:
Renamed as suggested.